app/Commands/Compile.hs

```diff
@@ -0,0 +1,178 @@
+          <> short 'o'
+          <> metavar "OUTPUT_FILE"
+          <> help "Path to output file"
+          <> action "file"
+          <> value "out"
+      )
+  _compileInputFile <- parserInputFile
+
+  pure CompileOptions {..}
+  where
+    parseTarget :: String -> Either String CompileTarget
+    parseTarget = \case
+      "wasm" -> Right TargetWasm
+      "c" -> Right TargetC
+      s -> Left $ "unrecognised target: " <> s
+
+    targetShow :: CompileTarget -> String
+    targetShow = \case
+      TargetC -> "c"
+      TargetWasm -> "wasm"
+
+    parseRuntime :: String -> Either String CompileRuntime
+    parseRuntime = \case
+      "standalone" -> Right RuntimeStandalone
+      "libc" -> Right RuntimeLibC
+      s -> Left $ "unrecognised runtime: " <> s
+
+    runtimeShow :: CompileRuntime -> String
+    runtimeShow = \case
+      RuntimeStandalone -> "standalone"
+      RuntimeLibC -> "libc"
+
+inputCFile :: CompileOptions -> FilePath
+inputCFile o = minijuvixBuildDir </> outputMiniCFile
+  where
+    outputMiniCFile :: FilePath
+    outputMiniCFile = takeBaseName (o ^. compileInputFile) <> ".c"
+
+runCompile :: CompileOptions -> Text -> IO (Either Text ())
+runCompile o minic = do
+  createDirectoryIfMissing True minijuvixBuildDir
+  TIO.writeFile (inputCFile o) minic
+  prepareRuntime o
+  case o ^. compileTarget of
+    TargetWasm -> clangCompile o
+    TargetC -> return (Right ())
+
+prepareRuntime :: CompileOptions -> IO ()
+prepareRuntime o = do
+  mapM_ writeRuntime runtimeProjectDir
+  where
+    standaloneRuntimeDir :: [(FilePath, BS.ByteString)]
+    standaloneRuntimeDir = $(FE.makeRelativeToProject "minic-runtime/standalone" >>= FE.embedDir)
+    libCRuntimeDir :: [(FilePath, BS.ByteString)]
+    libCRuntimeDir = $(FE.makeRelativeToProject "minic-runtime/libc" >>= FE.embedDir)
+    runtimeProjectDir :: [(FilePath, BS.ByteString)]
+    runtimeProjectDir = case o ^. compileRuntime of
+      RuntimeStandalone -> standaloneRuntimeDir
+      RuntimeLibC -> libCRuntimeDir
+    writeRuntime :: (FilePath, BS.ByteString) -> IO ()
+    writeRuntime (filePath, contents) =
+      BS.writeFile (minijuvixBuildDir </> takeFileName filePath) contents
+
+clangCompile :: CompileOptions -> IO (Either Text ())
+clangCompile o = do
+  v <- sysrootEnvVar
+  case v of
+    Left s -> return (Left s)
+    Right sysrootPath -> withSysrootPath sysrootPath
+  where
+    sysrootEnvVar :: IO (Either Text String)
+    sysrootEnvVar = maybeToEither "Missing environment variable WASI_SYSROOT_PATH" <$> lookupEnv "WASI_SYSROOT_PATH"
+    withSysrootPath :: String -> IO (Either Text ())
+    withSysrootPath sysrootPath = runClang clangArgs
+      where
+        clangArgs :: [String]
+        clangArgs = case o ^. compileRuntime of
+          RuntimeStandalone -> standaloneArgs sysrootPath outputFile inputFile
+          RuntimeLibC -> libcArgs sysrootPath outputFile inputFile
+        outputFile :: FilePath
+        outputFile = o ^. compileOutputFile
```

Comment:
Imagine that I simply run `minijuvix compile Input.mjuvix`. Can we generate a file named `Input.X` instead of simply `out`? where X is wasm or c depending on the default value.

Reply:
I did this in 7d87d3b6a7d4c11a0efd1709f78516f6fcc20bd3